CMSgov/design-system
# WNMGDS-1204 dialog icon prop

## Summary
Updating the Dialog component to have an icon prop that allows for a custom close icon

### Added
* `closeIcon` prop for Dialog component that is defaulted to the <CloseIcon />
* In medicare, defaulted the new prop to <CloseIconThin />

### Removed
* Dialog close icon background-image

### Changed
* Dialog documentation HTML examples to reflect the changes

## How to test
Run `yarn start` to build the core doc site. Compare [local Modal Dialog](http://localhost:3000/components/dialog/) page to [production](https://design.cms.gov/components/dialog/). Confirm the dialog close buttons look the same

Run `yarn start:medicare` to build the mgov doc site. Compare to [prod site](https://cmsgov.github.io/design-system/medicare/components/dialog/). Confirm close button looks the same.

@ericwbailey, I ran Axe and didn't see any errors with updating the close button like this, but please let me know if anything stand out to you accessibility-wise.
